### PR TITLE
feat: added clipboard api

### PR DESCRIPTION
**Pull request by jessekelly881**
Wraps the clipboard api. Useful in the browser as well as in other platforms(see below libraries). 

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard
https://github.com/sindresorhus/clipboardy
https://github.com/rsp/deno-clipboard

### CODE REVIEW

**Comment by tim-smart** [comment]
It would be nice to support attachments/binary too. Maybe a ClipboardItem data structure?

**Comment by jessekelly881** [comment]
Agreed. Let me take a look at adding it.

**Comment by jessekelly881** [comment]
So, adding this for browser is easy enough but all of the node/deno "copy-to-clipboard" libraries that I've seen only support text(not binary) and rely on piping it to a cli command. I don't know if it's possible to do this with binary data.

**Comment by jessekelly881** [comment]
I don't think this is possible in general sense on platform this relies on piping the input the specific cli tools. Unless we find a way around this what do you think should be given priority: supporting node(or bun/deno) or allowing passing binary data to the clipboard?

**Comment by tim-smart** [comment]
OK, for now you can remove it from the base /platform package and just keep it in platform-browser

File: docs/platform-browser/Clipboard.ts.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+---
+
+<h2 class="text-delta">Table of contents</h2>
+
+- [constructor](#constructor)
+  - [make](#make)
+- [interface](#interface)
+  - [Clipboard (interface)](#clipboard-interface)
+- [layers](#layers)
+  - [layerLive](#layerlive)
```
Comment (tim-smart) [inline docs/platform-browser/Clipboard.ts.md:20]: This should be `layer` to be consistent with the other modules.

File: docs/platform-browser/Clipboard.ts.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+# interface
+
+## Clipboard (interface)
+
+**Signature**
+
+```ts
+export interface Clipboard {
+  read: Effect.Effect<never, PlatformError, ClipboardItems>
```
Comment (tim-smart) [inline docs/platform-browser/Clipboard.ts.md:46]: Add `readonly` modifiers here to match the other modules.

File: packages/platform-browser/src/internal/clipboard.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import { Tag } from "@effect/data/Context"
+import * as Effect from "@effect/io/Effect"
+import * as Layer from "@effect/io/Layer"
+import type * as Clipboard from "@effect/platform-browser/Clipboard"
+import * as PlatformError from "@effect/platform/Error"
+
+/** @internal */
+export const tag = Tag<Clipboard.Clipboard>("@effect/platform/Clipboard")
```
Comment (tim-smart) [inline packages/platform-browser/src/internal/clipboard.ts:8]: s/platform/platform-browser/

File: packages/platform-browser/src/Clipboard.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  writeBlob: (blob: Blob) => Effect.Effect<never, PlatformError, void>
+  clear: Effect.Effect<never, PlatformError, void>
+}
+
+/**
+ * @since 1.0.0
+ * @category constructor
+ */
+export const make: (
+  impl: Omit<Clipboard, "clear">
```
Comment (tim-smart) [inline packages/platform-browser/src/Clipboard.ts:29]: This needs to be updated to match the internal signature.

**Review by tim-smart — CHANGES_REQUESTED** [review]

**Review by tim-smart — APPROVED** [review]

**Pull request merged**